jurity/recommenders/ips.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+# Copyright FMR LLC <opensource@fidelity.com>
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import List, Union, Tuple, Optional
+
+import numpy as np
+import pandas as pd
+
+from jurity.recommenders.base import _BaseRecommenders
+from jurity.utils import Constants, get_sorted_clicks
+
+
+class IPS(_BaseRecommenders):
+    """Inverse Propensity Score
+
+    Calculates the IPS, an estimate of CTR with a weighted correction based on how likely an item was to be recommended
+    by the historic policy if the user saw the item in the historic data.
+
+    ..math::
+        IPS = \frac{1}{n} \sigma \frac{r_a \times I(\hat{a} = a}{p(a|x,h)}
+
+    In this equation:
+    * n is the total size of the test data
+    * r_a is the observed reward
+    * hat{a} is the recommended item
+    * I(\hat{a} = a} is a boolean of whether the user-item pair has historic data
+    * p(a|x,h) is the probability of the item being recommended for the test context given the historic data
+
```

Comment:
same comment (great details! can also add a paper reference, if there is a clear/standard paper that goes with IPS in the literature)

Reply:
The DR reference can serve for both since that is the specific paper I used for this

Reply:
If historic policy is the same as current policy, CTR=IPS, right?

Reply:
Yes IPS is for when the "policy" or model is different, so the likelihood of an item being recommended is different